python/examples/core/harness.py

```diff
@@ -229,6 +229,56 @@ def _run_benchmark_n_iters(callback: Callable[[int], None], n_iters: int,
   return np.asarray([_pytimed(callback, *args) for _ in range(n_iters)])
 
 
+def _parse_problem_sizes(argument: str) -> Sequence[Union[int, Sequence[int]]]:
```

Comment:
Seems like this could be implemented much more simply with comprehensions?

Some cleaned/compressed version of:
```
cur, next = [elem for elem in c.split(',')], []

# At 
changed = True
while changed:
  changed = False

  for c in cur:
    if type(c) == str and c.startswith('['):
      next.append([elem for elem in c[1:-1].split(',')])
      changed = True
    else:
      next.append(int(c))

  cur = next

return cur
```

Reply:
yep will try to simplify this as discussed offline!